src/network/plugins/base/base-network-plugin.ts

```diff
@@ -47,9 +49,27 @@ export default class BaseNetworkPlugin extends Events {
         throw `Invalid string PeedID [${peerId}]: ${e.message}`;
       }
     }
+
+
     try {
       let peer: Libp2pPeer = await this.network.libp2p.peerStore.get(peerId)
         .catch(e => null)
+
+      if(peer && !peer.metadata.get("timestamp")){
+        const timestamp = Date.now();
+        const uint8Array = numToUint8Array(timestamp);
+        try{
+          this.network.libp2p.peerStore.patch(peerId, {
+            metadata: {timestamp: uint8Array}
+          });
+        }catch(e){
+          this.defaultLogger.error(`cannot patch peerStore, ${e.message}`);
+        }
+      }
+
+      peer = this.validatePeerstoreTimestamp(peer);
```

Comment:
What's the purpose of this fun?
Can't it be replaced by a simple condition?
Is it necessary to delete the peer from peerStore. Can't we get the new data from peerRouting and patch it?